### DISCUSSION TITLE
#split with left/right windows on seems to buffer and dump output.

### DISCUSSION THREAD

**Discussion by thechanster** [scandum/tintin#146] [General]
Hi All,

Testing out more advanced #split scenarios, but every time I enable a left or right split, the scrolling region seems to buffer output and dump it all out in one go. e.g, with the #split on, I speedwalk 5n - seems to output a single line, pause output/buffer the rest of the 5n, and dump it out all at once. Rather then the expected behavior of smoothly outputting the rooms as I walk through them, which happens with no split windows left or right of the scrolling window.

Hope that makes sense - Any suggestion?

Regards,
Matt

**Comment by scandum** [comment]
Hiyas,

Unfortunately that's the only way to get vertical splitting working as far as I know. I could implement a slow scroll, but that would send a lot of data to the terminal and likely result in flicker. 
 
It might also just be a matter of getting used to it? The current screen refresh rate is 10 times per second, which should work for most MUDs. 

If you're comfortable with C you could edit update.c and change
```
#define PULSE_UPDATE_PACKETS            10
```
to
```
#define PULSE_UPDATE_PACKETS            5
```
Then recompile, and see if that improves the situation. Let me know how that goes if you do.

**Comment by thechanster** [comment]
Thanks for getting back - screen refresh 10 times / s would work - However I'm not getting output for varying periods of time, from 2s to up to 10s, so we're talking possibly hundreds of lines of output at a time. Sounds like that is not expected?

**Comment by scandum** [comment]
It's not, are you connected over ssh? If so looking into enabling compression for ssh might help if it's not automatically enabled.

**Comment by thechanster** [comment]
Thanks again, I gave it a try - connecting to Centos 8.x box from Windows using SecureCRT. Turned on compression, no change, tested from Putty with/without compression, no change, tested to another tintin++ instance on Centos 7.x, no change.  Very odd. Was hoping this could replace my tmux setup. Appreciate your help so far.

**Comment by scandum** [comment]
Do you have the same 2-10 second delays with tmux as well?

I could maybe have it scroll at 20 pages a second. Hard to tell whether that'll be tricky or not.

**Comment by thechanster** [comment]
No delays, smooth scrolling in the tintin++ panel of tmux (only have input split, no side splits). All other tmux panels are running bash commands. Feel like the scroll independently, not really sure.

**Comment by scandum** [comment]
It sounds like something weird is going on. One thing to try, make sure you're running the latest version. Also, does it make a difference if you're using vertical splits inside and outside of tmux?

If you need to detach your session, tintin comes with its own detach support, see #help daemon. It's still a bit rough around the edges, but functional.

**Comment by thechanster** [comment]
Yes, running latest version.

Just some further comments, the scrolling window is the only one having issues.
I have status data overlaying the split line, and it's updating as I speedwalk. I also open a map window in the right split, I speedwalk and I move around the map at the expected rates, while the scrolling window is paused!

Tried vertical splits in tmux and same problem.

**Reply by scandum** [reply]
I've made a note to do some testing. What are your exact #split settings? Does it happen with a clean session and only the #split set?

Also, do you have an unusual #config packet_patch setting?

**Comment by thechanster** [comment]
Hey scandum - 

So I think I've resolved the issue - I am 100% sure I tested this with a clean session yesterday. However after your post above I felt it prudent to confirm once more before getting back to you. Found that it seems to flow smooth from clean session now. Loading my full script set, caused the problem to return. I checked and did have #CONFIG {PACKET PATCH} {0.28}. Removing this / setting to 0 seems to resolve the issue. 

Sorry to waste your time, I am certain I tested with clean session so I am a bit baffled, but hopefully it's all good from here. Really appreciate your help though.

Matt

**Comment by scandum** [comment]
Thanks, this definitely seems like a bug, at least I now know how to go about reproducing it!
